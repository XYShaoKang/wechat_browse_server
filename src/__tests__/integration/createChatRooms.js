import { Prisma, graphqlTestCall } from '../__utils'
import { CreateChatRooms } from '../__types'
import { chatRoomData } from '../__variables'

describe.only('Mutation', () => {
  /** @type {Prisma} */
  let prisma
  beforeEach(() => {
    prisma = new Prisma()
  })
  it('create ChatRoom', async () => {
    const result = await graphqlTestCall({
      query: CreateChatRooms,
      variables: chatRoomData,
      context: {
        prisma,
        currentUser: { id: '0' },
        currentWeChat: { id: '0' },
      },
    })

    const id = result.data && result.data.CreateChatRooms.id

    expect(id).toBe('0')
    expect(prisma.updateWeChat.mock.calls.length).toBe(1)
    expect(result).toMatchSnapshot()
    expect(prisma.db.chatRooms).toMatchSnapshot()
    expect(prisma.db.avatars).toMatchSnapshot()
  })
  it('create WeChatUser Fail No WeChat', async () => {
    const result = await graphqlTestCall({
      query: CreateChatRooms,
      variables: chatRoomData,
      context: { prisma, currentUser: { id: '0' } },
    })
    const message = result.errors && result.errors[0].message

    expect(message).toBe('No WeChat')
    expect(result).toMatchSnapshot()
  })
})
