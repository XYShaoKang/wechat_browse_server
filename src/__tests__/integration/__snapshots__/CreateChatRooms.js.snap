// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation create ChatRoom 1`] = `
Object {
  "data": Object {
    "CreateChatRooms": Object {
      "id": "0",
    },
  },
}
`;

exports[`Mutation create ChatRoom 2`] = `
Array [
  Object {
    "displayName": "",
    "id": "0",
    "memberList": Array [
      Object {
        "id": "0",
      },
      Object {
        "id": "1",
      },
    ],
    "modifyTime": "2015-04-24T16:51:14.743Z",
    "nickname": "",
    "owner": Object {
      "id": "0",
    },
    "username": "b@chatroom",
  },
  Object {
    "avatar": Object {
      "id": "0",
    },
    "displayName": "cccc",
    "id": "1",
    "memberList": Array [
      Object {
        "id": "0",
      },
      Object {
        "id": "1",
      },
    ],
    "modifyTime": 2019-04-24T16:51:14.743Z,
    "nickname": "bbbb",
    "owner": Object {
      "connect": Object {
        "id": "0",
      },
    },
    "username": "aaaa@chatroom",
  },
]
`;

exports[`Mutation create ChatRoom 3`] = `
Array [
  Object {
    "bigImg": Object {
      "id": "1",
    },
    "id": "0",
    "thumbnailImg": Object {
      "id": "0",
    },
  },
]
`;

exports[`Mutation create WeChatUser Fail No WeChat 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: No WeChat],
  ],
}
`;
